### DISCUSSION TITLE
accessing published streams codec and other info

### DISCUSSION THREAD

**Discussion by scottf-tvw** [bluenviron/mediamtx#2904] [Questions]
### Question

I am looking to get information about the published stream such as codec, bitrate, frame-rate, dims ect.. it would be nice if that came along with an API query output for paths -> source? 
and or as part of the vars that can be used to send hooks (might be hard to get valuable info right after a publish however) 
is this data somewhere already?

**Comment by aler9** [comment]
Hello, the API already provides the codec of each published track.

Regarding additional informations such as framerate, frame size, they are codec-dependent, therefore they are not trivial to extract. I've created a feature request here: #2955

**Reply by aler9** [reply]
additional info: the bitrate is provided too - it can be inferred by deriving bytesReceived and bytesSent.
